### ISSUE TITLE
ROCm 2.10: rocm-smi --setsclk  does not stick to passed value

### ISSUE THREAD

**Issue by valeriob01** [RadeonOpenCompute/ROCm#1051]
setting sclk does not stick to passed value, but frequency varies over time.

**Comment by selroc** [comment]
> setting sclk does not stick to passed value, but frequency varies over time.

setting sclk 6 never sticks to 1801 MHz but to a lower frequency. Suggestions?

**Comment by valeriob01** [comment]
No luck with setting gpu by hand:
echo "vc 2 1801 950" > /sys/class/drm/card0/device/pp_od_clk_voltage
bash: /sys/class/drm/card0/device/pp_od_clk_voltage: Permission denied

**Comment by valeriob01** [comment]
Conversely --setmclk always sticks to the passed value.

**Comment by notokcomputerbadcomputer** [comment]
TL;DR, it seems driver accounts for power dissipation and safe temperatures regardless of user setting.

I'm running different cards at manual with gpu and memory overrides. In a workroom with HVAC set to 18 degrees C. They do stick to override values. However when HVAC is off the gpu clock starts to fluctuate and if i want to work in comfortable 25 degree C, the gpu downclocks significantly, as low as level 3 or 4.
Memory doesn't seem affected as much, so it sticks to settings in all conditions.

**Comment by valeriob01** [comment]
> TL;DR, it seems driver accounts for power dissipation and safe temperatures regardless of user setting.
> 
> I'm running different cards at manual with gpu and memory overrides. In a workroom with HVAC set to 18 degrees C. They do stick to override values. However when HVAC is off the gpu clock starts to fluctuate and if i want to work in comfortable 25 degree C, the gpu downclocks significantly, as low as level 3 or 4.
> Memory doesn't seem affected as much, so it sticks to settings in all conditions.

Thank you for the explanation.

**Comment by valeriob01** [comment]
> > TL;DR, it seems driver accounts for power dissipation and safe temperatures regardless of user setting.
> > I'm running different cards at manual with gpu and memory overrides. In a workroom with HVAC set to 18 degrees C. They do stick to override values. However when HVAC is off the gpu clock starts to fluctuate and if i want to work in comfortable 25 degree C, the gpu downclocks significantly, as low as level 3 or 4.
> > Memory doesn't seem affected as much, so it sticks to settings in all conditions.
> 
> Thank you for the explanation.

however it is too early for me to turn on the HVAC, I have to continue without for some months.
Thank you again :-)
